src/gui/plugins/component_inspector/ComponentInspector.cc

```diff
@@ -1267,6 +1270,42 @@ transport::Node &ComponentInspector::TransportNode()
   return this->dataPtr->node;
 }
 
+/////////////////////////////////////////////////
+void ComponentInspector::QuerySystems()
+{
+  msgs::Empty req;
+  msgs::EntityPlugin_V res;
+  bool result;
+  unsigned int timeout = 5000;
+  std::string service{"/world/" + this->dataPtr->worldName +
+      "/entity/system/info"};
+  if (!this->dataPtr->node.Request(service, req, timeout, res, result))
+  {
+    ignerr << "Unable to query available systems." << std::endl;
+    return;
+  }
+
+  this->dataPtr->systemFilenameList.clear();
+  for (auto &plugin : res.plugins())
+  {
+    this->dataPtr->systemFilenameList.push_back(
```

Comment:
hmm I tried a few different approaches but no luck yet. Looks like instead of using a simple QStringList, we may have to create a model and populate it in C++ with text and value pair so the text can be displayed differently. This is still a todo item.

Reply:
I went ahead and kept the mapping in a separate variable for now: dca30043248fdf1a7670ec7cd34fb98e39bc0328

Reply:
oh nice thanks for tackling this! I was fighting with QML for a long time to try get it working.